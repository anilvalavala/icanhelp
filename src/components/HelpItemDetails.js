import React from 'react';
import { Link } from 'react-router-dom';

const HelpItemDetails = (props) => {
    return(
        <div>
            <div>
                <Link to={`/editHelp/${props.helpItem.id}`}> {props.helpItem.title} </Link> <br />
                Description: {props.helpItem.description} <br />
                Available From: {props.helpItem.fromDate} <br />
                Available Upto: {props.helpItem.toDate} <br />
                Email: {props.helpItem.email} <br />
                Phone: {props.helpItem.phone} <br />
                <button onClick={
                    () => {
                        props.dispatch(deleteHelpItem(props.helpItem.id));
                    }
                }>Delete</button>
            </div>
        </div>
    );
}; 

export default HelpItemDetails;