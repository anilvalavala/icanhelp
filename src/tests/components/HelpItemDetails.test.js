import React from 'react';
import { shallow } from 'enzyme';
import HelpItemDetails from '../../components/HelpItemDetails';
import TestHelpItems from '../fixtures/TestHelpItems';

test('Test HelpItemDetails component by passing a test help item', () => {
    const wrapper = shallow(<HelpItemDetails helpItem={TestHelpItems[1]}/>);
    expect(wrapper).toMatchSnapshot();
});

test('Test HelpItemDetails component without passing a test help item', () => {
    const wrapper = shallow(<HelpItemDetails helpItem={{}} />);
    expect(wrapper).toMatchSnapshot();
});
