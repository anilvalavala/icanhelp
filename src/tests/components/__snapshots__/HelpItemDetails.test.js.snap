// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test HelpItemDetails component by passing a test help item 1`] = `
<div>
  <div>
    <Link
      replace={false}
      to="/editHelp/2"
    >
       
      books
       
    </Link>
     
    <br />
    Description: 
    K.G books
     
    <br />
    Available From: 
    86400000
     
    <br />
    Available Upto: 
    259200000
     
    <br />
    Email: 
    anilkumar.v@hotmail.com
     
    <br />
    Phone: 
    9545947627
     
    <br />
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;

exports[`Test HelpItemDetails component without passing a test help item 1`] = `
<div>
  <div>
    <Link
      replace={false}
      to="/editHelp/undefined"
    >
       
       
    </Link>
     
    <br />
    Description: 
     
    <br />
    Available From: 
     
    <br />
    Available Upto: 
     
    <br />
    Email: 
     
    <br />
    Phone: 
     
    <br />
    <button
      onClick={[Function]}
    >
      Delete
    </button>
  </div>
</div>
`;
